operate/qa/integration-tests/src/test/java/io/camunda/operate/elasticsearch/FinishedImportingIT.java

```diff
@@ -172,6 +177,104 @@ final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
         .until(() -> isRecordReaderIsCompleted("2-process-instance"));
   }
 
+  @Test
+  public void shouldNotMarkedAsCompletedViaMetricsWhenImportingIsNotDone() throws IOException {
+    // given
+    final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    final var partitionTwoRecord = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 2);
+    EXPORTER.export(record);
+    EXPORTER.export(partitionTwoRecord);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    // when
+    for (int i = 0; i <= RecordsReaderHolder.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER; i++) {
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    // then
+    final MeterRegistry metrics = beanFactory.getBean(MeterRegistry.class);
+
+    final Gauge partitionOneImportStatus =
+        metrics
+            .get(Metrics.GAUGE_NAME_IMPORT_POSITION_COMPLETED)
+            .tags(
+                Metrics.TAG_KEY_PARTITION,
+                "1",
+                Metrics.TAG_KEY_IMPORT_POS_ALIAS,
+                "process-instance")
+            .gauge();
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(() -> assertThat(partitionOneImportStatus.value()).isEqualTo(0.0));
+
+    final Gauge partitionTwoImportStatus =
+        metrics
+            .get(Metrics.GAUGE_NAME_IMPORT_POSITION_COMPLETED)
+            .tags(
+                Metrics.TAG_KEY_PARTITION,
+                "2",
+                Metrics.TAG_KEY_IMPORT_POS_ALIAS,
+                "process-instance")
+            .gauge();
+    assertThat(partitionTwoImportStatus.value()).isEqualTo(0.0);
```

Comment:
@EuroLew you mean this test or?

Reply:
For the communication of importer being finished (Assuming each record reader has an entry in the import postion index from processing 8.6 records which it should have? I don't think I can guarantee this assumption for prod though) A specific record reader does not need messages for it to be marked as completed e.g.

1. process instance record reader receives an 8.7 record marks partition 1 as completed
2. job record reader for partition 1 never receives a record but because it knows the partition is complete and it gets 5 empty batches we can mark it as complete

however for these metrics solutions we will only mark a record reader as 1.0 or 0.0 IF we receive a record for that record reader, given the PR description says this is intended to know where importing stands and whether it's done, this could be a problem as there could be uncompleted record readers that do not show up in metrics as 0.0 (if the corresponding record reader does not receive a record).

Or maybe this is not a big deal, if we have it configured somewhere or stored somewhere the amount of record readers we have we can compare how many 1.0 record readers we have

Reply:
>however for these metrics, we will only mark a record reader as 1.0 or 0.0 IF we receive a record for that record reader, given the PR description says this is intended to know where importing stands and whether it's done, this could be a problem as there could be uncompleted record readers that do not show up in metrics as 0.0 (if the corresponding record reader has never received a record).

That is not true 🤔 It will register the first time it sets the import position the gauge https://github.com/camunda/camunda/pull/25038/commits/ba3d5edb021505abfa0aca2bc743d9820bcf2d77#diff-c2f26dd940eb71359693407274f4e401317be38aab4d7e51162e7ecafa775b27R132 

Here it keeps the reference of the import position entity. This can either be when we import more records or when we complete with empty batches, because we take the same path on marking as done.

Reply:
resolved

Reply:
I synced with @EuroLew we concluded that this should work, as the gauge is keeping track of the object reference and on marking reader complete we also run this code part